python/rikai/types/geometry.py

```diff
@@ -440,3 +448,207 @@ def to_dict(self) -> dict:
+        data: list[int]
+            the RLE data
+        height: int
+            The height of the image which the mask applies to.
+        width: int
+            The width of the image which the mask applies to.
+        """
+        return Mask(
+            data, height=height, width=width, mask_type=Mask.Type.COCO_RLE
+        )
+
+    @staticmethod
+    def from_polygon(data: list[list[float]], height: int, width: int) -> Mask:
+        """Build mask from a Polygon
+
+        Parameters
+        ----------
+        data: list[list[float]]
+            Multiple Polygon segmentation data. i.e.,
+            ``[[x0, y0, x1, y1, ...], [x0, y0, x1, y1, ...]])``
+        height: int
+            The height of the image which the mask applies to.
+        width: int
+            The width of the image which the mask applies to.
+        """
+        return Mask(
+            data, height=height, width=width, mask_type=Mask.Type.POLYGON
+        )
+
+    @staticmethod
+    def from_mask(mask: np.ndarray) -> Mask:
+        """Build mask from a numpy array.
+
+        Parameters
+        ----------
+        mask : np.ndarray
+            A binary-valued (0/1) numpy array
+        """
+        assert len(mask.shape) > 1, "Must have more than 2-dimensions"
+        return Mask(data=mask, mask_type=Mask.Type.MASK)
+
+    def __repr__(self):
+        return f"Mask(type={self.type}, data=...)"
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, Mask)
+            and self.type == other.type
+            and self.height == other.height
+            and self.width == other.width
+            and np.array_equal(self.data, other.data)
+        )
+
+    def _polygon_to_mask(self) -> np.ndarray:
+        arr = np.zeros((self.height, self.width), dtype=np.uint8)
+        with Image.fromarray(arr) as im:
+            draw = ImageDraw.Draw(im)
+            for polygon in self.data:
+                draw.polygon(list(np.array(polygon)), fill=1)
+            im = im.resize((self.width, self.height))
+            return np.array(im)
+
+    def to_mask(self) -> np.ndarray:
+        """Convert this mask to a numpy array."""
+        if self.type == Mask.Type.MASK:
+            return self.data
+        elif self.type == Mask.Type.POLYGON:
+            return self._polygon_to_mask()
+        elif self.type == Mask.Type.RLE:
+            return rle.decode(self.data, shape=(self.height, self.width))
+        elif self.type == Mask.Type.COCO_RLE:
+            return rle.decode(
+                self.data, shape=(self.height, self.width), order="F"
+            )
+        else:
+            raise ValueError("Unrecognized type")
+
+    def to_numpy(self) -> np.ndarray:
+        return self.to_mask()
+
+    def to_dict(self) -> dict:
+        d = {
+            "type": self.type.value,
+            "width": self.width,
+            "height": self.height,
+        }
+        if self.type == Mask.Type.RLE:
+            d["rle"] = self.data
```

Comment:
so why call to_mask if it's COCO_RLE but not regular RLE?

Reply:
A mistake. Fixed.